docs/4.4/api-reference.md

```diff
@@ -796,14 +798,129 @@ if err := client.DeleteAccessRequest(ctx, accessReqID); err != nil {
+
+**Automated Rotation**
+
+You can use `tctl auth rotate --type=user --grace-period=10h` or the following RPC to start the rotation in `auto` mode.
 
 ```go
-ca, err := client.GetCertAuthority(services.CertAuthID{
-  DomainName: clusterName,
-  Type:       services.HostCA,
-}, false)
-if err != nil {
+// This will start an automatic rotation that schedules each phase of the rotation in equal increments (each 1/3 of the grace period).
+gracePeriod := time.Hour * 24
+req := auth.RotateRequest{
+  Mode: services.RotationModeAuto,
+  GracePeriod: &gracePeriod, // defaults to 48 hours
+  Type: services.UserCA, // Leave empty to target UserCA and HostCA
+}
+if err := client.RotateCertAuthority(req); err != nil {
+  return err
+}
+```
+
+The grace period should be set to 2-3 times the expected time to rotate the CA to ensure completion, while minimizing the grace period. The `type` flag is useful if you want to rotate the user and host CA's with different strategies or frequencies.
+
+**Automated Rotation with a Custom Schedule**
+
+You can also rotate certificates with a custom schedule. You may want to extend the length of one phase in the schedule without having a ridiculously long grace period, since long grace periods present a possible vulnerability.
+
+```go
+// This will automate your CA rotation with the custom schedule. Use with caution, each phase should have extra time to ensure they complete in less than optimal situations.
+if err := client.RotateCertAuthority(auth.RotateRequest{
+  Mode: services.RotationModeAuto,
+  Schedule: &services.RotationSchedule{
+    // 1 hour for Init
+    UpdateClients: time.Now().UTC().Add(time.Hour),
+    // 4 hours for UpdateClients
+    UpdateServers: time.Now().UTC().Add(time.Hour * 5),
+    // 2 hours for UpdateServers
+    Standby: time.Now().UTC().Add(time.Hour * 7),
+  },
+}); err := client.RotateCertAuthority(req); err != nil {
   return err
 }
 ```
+
+**Manual Rotation** (custom automated solution)
+
+You can set up a custom system for rotation by triggering each phase manually with `manual` mode. This can be done with `tctl auth rotate --phase=update_clients` or the following RPC.
+
+```go
+// You can run this RPC to start the Update Servers phase
+if err := client.RotateCertAuthority(auth.RotateRequest{
+  Mode:        services.RotationModeManual,
+  TargetPhase: services.RotationPhaseUpdateServers,
+}); err != nil {
+  return err
+}
+```
+
+This can be used to make rotations independent of an arbitrary rotation schedule or grace period. For example, you might be able to set up a custom automated solution where each phase is triggered by the event of the prior phase completing. This is possible in theory, though it may be complicated to set up.
+
+However there are multiple upsides if you make it work:
+- it would be quicker than `auto` mode since it wouldn't wait for a phase that is already complete
```

Comment:
Reading this makes me wonder why `auto` is not implemented with events as described since it sounds like a more fitting solution.

Reply:
I'm also unsure of the reason, but my assumption is that this would be a pretty opinionated solution, and so it is left up to the customer to decide their strategy. It is also possible that this is too complicated/involved, and there simply was no time for it, or it is not common practice.